### ISSUE TITLE
a small bug in resnet model _make_layer() implementation

### ISSUE THREAD

**Issue by dakjin** [pytorch/vision#706]
Hi guys,

I think there is a small bug in the "_make_layer(self, block, planes, blocks, stride=1)" function (in charge of generating residual blocks at certain resolution) in the ResNet model.  This function will just simply discard the odd number of rows and columns in the feature maps in the identity path of a residual block when the stride=2 (the case where feature maps get downsampled by 2).  

    def _make_layer(self, block, planes, blocks, stride=1):
        downsample = None
        if stride != 1 or self.inplanes != planes * block.expansion:
            downsample = nn.Sequential(
                #this 1x1 conv with stride 2 will simply discard the odd number of rows and columns in the feature maps in the identity path
                **conv1x1(self.inplanes, planes * block.expansion, stride)**, 
                nn.BatchNorm2d(planes * block.expansion),
            )

Please let me know if I understand correctly about this issues.

**Comment by fmassa** [comment]
Hi,

Can you expand on what you think the problem is, and what would be a solution? I'm not sure I completely followed your question.

**Comment by ekagra-ranjan** [comment]
@dakjin I think I got your issue.  Correct me if I got it wrong.

The 1x1 conv block with stride = 2 skips rows and columns which make the computing of the skipped rows and columns useless as they are never used in computing the values for the next layer because the spatial coverage is only 1x1 (unlike 3x3 or 5x5 where the neighbouring values participate in the computation of the next layer). You might be thinking of an optimisation here by not computing the rows and columns which will be skipped in the first place. It might work had it been a sequential model. The catch here is that you cannot skip computing any rows and columns in the previous layer of 1x1 because the data follows a dual path and is needed by 3x3 kernels in the BasicBlock and Bottleneck!  So although some rows and columns might be useless for the 1x1 layers but are required for the parallel 3x3 layers.
